src/test/java/com/google/sps/servlets/StoreVideosTest.java

```diff
@@ -0,0 +1,94 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ConcurrentModificationException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+
+public final class StoreVideosTest {
+
+  private final LocalServiceTestHelper helper =
```

Comment:
Is this actually doing anything in the test since I don't see it being called or injected? or does it "magically" make the calls to datastore methods in the source code somehow write in-memory similar to like a fake/mock? I'm guessing that if the tests pass even after running multiple, then this is working but it just seems like magic to a read haha.

Reply:
LOL I'm not too sure how it actually works but yeah, it does "magically" configure datastore to write in-memory. I was pretty shocked when I read the example code as well lol